tempodb/encoding/vparquet3/block_traceql.go

```diff
@@ -39,8 +47,112 @@ type span struct {
 	cbSpanset      *traceql.Spanset
 }
 
-func (s *span) Attributes() map[traceql.Attribute]traceql.Static {
-	return s.attributes
+func (s *span) AllAttributes() map[traceql.Attribute]traceql.Static {
+	atts := make(map[traceql.Attribute]traceql.Static, len(s.spanAttrs)+len(s.resourceAttrs)+len(s.traceAttrs))
```

Comment:
If I'm not mistaken `AllAttributes()` is currently only called once per span. I'm still wondering if it makes sense to cache `atts` in the span and skip the copying for consecutive calls to avoid future performance issues